### PR TITLE
Remove build scripts for RPM and Debian packages

### LINKED ISSUE #69: RFC: remove packaging from upstream
With the Debian packaging initiative making progress, I think it is time to maybe reconsider the future of our in-repo packaging.

Debian packagers have reported in the past that having a "debian" directory makes their work harder when merging the upstream branch into their packaging branch. Moreover, our in-repo Debian packaging is less complete than the "soon to be ready" Debian packaging. The same remark may apply to our in-repo RPM packaging.

Maybe it is time to drop our in-repo packaging and invite people to install the targetcli-fb package from their distro in our README.md. We could replace the "in-repo packaging" section with a list of distributions that ship targetcli-fb (RHEL, Fedora, Suse, Archlinux, and soon Debian).

What do you think?

### PR DESCRIPTION
**Pull request by cvubrugier**
Remove the "debian" directory because Debian packagers have reported in
the past that having a "debian" directory makes their work harder when
merging the upstream branch into their packaging branch. Moreover, our
in-repo Debian packaging is less complete than the Debian packaging.

Also remove our in-repo RPM packaging which is probably lagging behind
Fedora's packaging.

Finally, this patch adds a few links to RPM and Debian build scripts in
our README.md.

Fixes #69.

Signed-off-by: Christophe Vu-Brugier <cvubrugier@fastmail.fm>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,15 +18,16 @@ targetcli-fb is licensed under the Apache 2.0 license. Contributions are welcome
...
-custom packaging easier -- distributions tend to maintain their own packaging
-scripts separately. If you run into issues with packaging, start with opening
-a bug on your distro's bug reporting system.
-
-Some people do use these scripts, so we want to keep them around. Fixes for
-any breakage you encounter are welcome.
+Packages
+--------
+targetcli-fb is packaged for a number of Linux distributions including
+RHEL, Fedora, openSuse, Arch Linux or Debian.
```
Comment (agrover) [inline README.md]: should read "Arch Linux, and Debian". (reason: "Oxford Comma" rule)

File: README.md [hunk 2]
```diff
@@ -18,15 +18,16 @@ targetcli-fb is licensed under the Apache 2.0 license. Contributions are welcome
...
+
+If you need to build a custom RPM or Debian package, you can refer to
+the build scripts from
+[Fedora](http://pkgs.fedoraproject.org/cgit/rpms/targetcli.git)
+or
+[Debian](https://alioth.debian.org/plugins/scmgit/cgi-bin/gitweb.cgi?p=linux-target/targetcli-fb.git).
```
Comment (agrover) [inline README.md]: @gonzoleeman will Fedora scripts work on SuSE or should we also include a link to SuSE-style script, for custom pkg builders?

**Comment by cvubrugier** [comment]
@agrover, I have amended my patch to follow the "Oxford Comma" rule.  Thank you for the grammar lesson :+1:

**Comment by gonzoleeman** [comment]
I concur with removing them, but examples are good, too. I used the SPEC file template to figure out how to build the RPM SPEC file for openSUSE. It would have been much harder to write it from scratch.

**Comment by agrover** [comment]
@gonzoleeman ok cool. I think a new distro looking to package these could look at the Fedora and Debian examples, so I think we're covered. (BTW additional links to distro script examples would also be welcome.)

**Comment by diogocp** [comment]
Gentoo: https://packages.gentoo.org/packages/sys-block/targetcli-fb

**Comment by cvubrugier** [comment]
@diogocp, I created pull requests to mention Gentoo and point to the Gentoo package URL for rtslib-fb, configshell-fb and targetcli-fb.

**Pull request merged**